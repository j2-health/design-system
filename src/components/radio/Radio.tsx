import {
  Radio as AntdRadio,
  RadioProps as AntdRadioProps,
  RadioGroupProps as AntdRadioGroupProps,
} from 'antd'

type RadioProps = Expand<AntdRadioProps>

type FullRadioGroupProps = AntdRadioGroupProps & {
  buttonStyle?: 'solid' | 'outline'
  optionType?: 'default' | 'button'
}
type RadioGroupProps = Expand<FullRadioGroupProps>

const Radio = (props: RadioProps) => {
  return <AntdRadio {...props} />
}

const RadioGroup = ({
  buttonStyle = 'solid',
  optionType = 'button',
  ...props
}: RadioGroupProps) => {
  return (
    <AntdRadio.Group
      {...props}
      buttonStyle={buttonStyle}
      optionType={optionType}
    />
  )
}

export { Radio, RadioGroup }
