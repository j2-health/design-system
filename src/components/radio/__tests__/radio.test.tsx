import { render } from '@testing-library/react'
import { Radio, RadioGroup } from '../Radio'

describe('Radio', () => {
  it('should render correctly', () => {
    const { container } = render(<Radio value="aetna" />)
    expect(container).toMatchSnapshot()
  })
})

describe('RadioGroup', () => {
  it('should render correctly', () => {
    const { container } = render(<RadioGroup options={['aetna', 'cigna']} />)
    expect(container).toMatchSnapshot()
  })
})
