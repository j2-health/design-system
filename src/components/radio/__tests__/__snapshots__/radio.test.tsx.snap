// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `
<div>
  <label
    class="ant-radio-wrapper css-dev-only-do-not-override-os2fvf"
  >
    <span
      class="ant-radio ant-wave-target"
    >
      <input
        class="ant-radio-input"
        type="radio"
        value="aetna"
      />
      <span
        class="ant-radio-inner"
      />
    </span>
  </label>
</div>
`;

exports[`RadioGroup should render correctly 1`] = `
<div>
  <div
    class="ant-radio-group ant-radio-group-solid css-dev-only-do-not-override-os2fvf"
  >
    <label
      class="ant-radio-button-wrapper css-dev-only-do-not-override-os2fvf"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          name="test-id"
          type="radio"
          value="aetna"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span
        class="ant-radio-button-label"
      >
        aetna
      </span>
    </label>
    <label
      class="ant-radio-button-wrapper css-dev-only-do-not-override-os2fvf"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          name="test-id"
          type="radio"
          value="cigna"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span
        class="ant-radio-button-label"
      >
        cigna
      </span>
    </label>
  </div>
</div>
`;
